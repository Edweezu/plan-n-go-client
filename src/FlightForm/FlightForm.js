import React from 'react'
import config from '../config'
import TokenService from '../services/token-service'
import TripsContext from '../TripsContext';


class FlightForm extends React.Component {

    static contextType = TripsContext

    handleAddFlight = (e) => {
        e.preventDefault()

        const tripid = this.props.tripid
        console.log('tripiddd', tripid)

        const { airline, flight_num, depart_date, depart_time, seats, flight_notes } = e.target

        const newFlight = {
            airline: airline.value,
            flight_num: flight_num.value,
            depart_date: depart_date.value,
            depart_time: depart_time.value,
            seats: seats.value,
            flight_notes: flight_notes.value,
        }

        console.log('newwwflight', newFlight)
        fetch(`${config.API_ENDPOINT}/trips/${tripid}/flights`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify(newFlight),
        })
        .then(res => {
            if (!res.ok)
            return res.json().then(e => Promise.reject(e))
            return res.json()
        })
        .then(flight => {
            console.log('posted flight', flight)
            airline.value = ''
            flight_num.value = ''
            depart_date.value = ''
            depart_time.value = ''
            seats.value = ''
            flight_notes.value = ''
            this.context.setFlightDone()
            this.context.addFlight(flight)
        })
        .catch(err => {
            console.error({ err })
        })
    }


    render () {
        return (
            <form onSubmit={this.handleAddFlight}>
                <div className="form-section">
                    <label htmlFor="airline">Airline *</label>
                    <input type="text" name="airline" id="airline" required/>
                </div>
                <div className="form-section">
                    <label htmlFor="flight_num">Flight #</label>
                    <input type="number" name="flight_num" id="flight_num"/>
                </div>
                <div className="form-section">
                    <label htmlFor="depart_date">Departure Date *</label>
                    <input type="date" name="depart_date" id="depart_date" required/>
                </div>
                <div className="form-section">
                    <label htmlFor="depart_time">Departure Time</label>
                    <input type="time" name="depart_time" id="depart_time"/>
                </div>
                <div className="form-section">
                    <label htmlFor="seats">Seats</label>
                    <input type="text" name="seats" id="seats"/>
                </div>
                <div className="form-section">
                    <label htmlFor="flight_notes">Notes</label>
                    <input type='text' name="flight_notes" id="flight_notes"
                    />
                </div>
                <button type="submit">Submit</button>
            </form>
        )
    }
}

export default FlightForm