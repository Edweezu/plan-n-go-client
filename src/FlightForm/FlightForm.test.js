import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import FlightForm from './FlightForm'


describe('FlightForm component', () => {
    const props = {
        tripid: 1
    }

    it('renders the given component', () => {
        const wrapper = shallow(<FlightForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<FlightForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})