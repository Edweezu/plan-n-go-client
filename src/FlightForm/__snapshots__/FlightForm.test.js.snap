// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightForm component renders the component's given props 1`] = `
<section
  className="trip-flight-form"
>
  <header
    className="flight-form-header"
  >
    <h2>
      Flights 
      <i
        className="fas fa-plus-circle"
        onClick={[Function]}
      />
    </h2>
  </header>
</section>
`;

exports[`FlightForm component renders the given component 1`] = `
<section
  className="trip-flight-form"
>
  <header
    className="flight-form-header"
  >
    <h2>
      Flights 
      <i
        className="fas fa-plus-circle"
        onClick={[Function]}
      />
    </h2>
  </header>
</section>
`;
